hooks/pre-command

```diff
@@ -40,16 +65,21 @@ maybe_download_sc() {
   case "${OSTYPE}" in
     linux*)
       os="linux"
-      # TODO: cache this download on the agent
-      curl -fsSL --retry 5 "https://saucelabs.com/downloads/sc-${sauce_connect_version}-${os}.tar.gz" | tar -xzf - "sc-${sauce_connect_version}-${os}/bin/sc"
+
+      local sc_tarball_path
+      sc_tarball_path=$(maybe_download_sc_archive "${sauce_connect_version}" "${os}")
+
+      tar -xzf "${sc_tarball_path}" "sc-${sauce_connect_version}-${os}/bin/sc"
```

Comment:
Since the extraction logic is different based on the OS, let's move it into `maybe_download_sc_archive` and rename that `maybe_download_sc`, so it can return the path to the executable instead of the archive. That way we don't need to switch based on os both inside and outside the function.

Reply:
c2afc277960b361b5f378debf4f739d42d547578

Reply:
Pulled the logic for generating the archive name out to `maybe_download_sc` so the caching logic can be shared, but we only have one switch now.